frontendFile/src/main/java/com/virtuslab/gitmachete/frontend/file/grammar/Machete.flex

```diff
@@ -0,0 +1,58 @@
+// Copyright 2000-2020 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.virtuslab.gitmachete.frontend.file.grammar;
+
+import com.intellij.lexer.FlexLexer;
+import com.intellij.psi.tree.IElementType;
+import com.intellij.psi.TokenType;
+
+%%
+
+%class MacheteLexer
+%implements FlexLexer
+%unicode
+%function advance
+%type IElementType
+%eof{  return;
+%eof}
+
+
+// End Of Line
+EOL=[\r\n]
+// For sake of distinguishablility WHITESPACE and INDENTATION are two separated entries
+// even if they are the same for now
+WHITESPACE=[\ \t]
+INDENTATION=[\ \t]
+// Probably all characters allowed in branch name
+// (https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git-check-ref-format.html)
+NAME_WITHOUT_SLASH=[\w\-.!@#$%&()+={}\];'\",.<>|`]
```

Comment:
There's an inconsistency in naming: `NAME_WITH{,OUT}*_SLASH` and `ANNOTATION` suggests a sequence of chars, while in fact they correspond just to a single char... maybe `NAME_CHAR_WITH{,OUT}_SLASH` and `ANNOTATION_CHAR`?

Reply:
(unless that's a {JF,F,}lex naming convention?)

Reply:
I changed this in a way that i.e. `ANNOTATION` is representation of one or many characters not single one.